test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Configuration.Test
+{
+    public class SearchTreeTests
+    {
+        [Fact]
+        public void SearchTree_WithOneSource()
```

Comment:
Follow test [class](https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#unit-test-class-naming) and [method](https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#unit-test-method-naming) naming conventions

Reply:
I agree. But in this case in same method we're doing several assertion some may be success, other could be failure. So I prefer not to name `SearchTree_WithOneSource_Success` this case.
It's same as below one.
https://github.com/NuGet/NuGet.Client/blob/42e68a74fce207a2b9c442886071f51d2c20ebe5/test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageNamespacesConfigurationTests.cs#L16

Reply:
You can use a theory instead. 

It's better to have as few asserts as possible. I think you can easily achieve that here.

Reply:
@dtivel  @nkolev92 
1. I renamed the test file.
2. Now creating `SearchTree` instance and testing against it.
3. I renamed them more accurately.
4. I separate negative and positive tests.
Implemented theory. But `SearchTree_TopNodeIsGlobbing_Match` is quite complex one so I left as it's.